dcpquery/etl/__init__.py

```diff
@@ -36,21 +32,83 @@ def transform_bundle(bundle_uuid, bundle_version, bundle_path, bundle_manifest_p
                                             content_type=fm["content-type"],
                                             name=fm["name"],
                                             size=fm["size"]))
+
+    combine_file_data(result)
     return result
 
 
-def load_bundle(bundle, extractor, transformer):
-    bf_links = []
-    bundle_row = Bundle(uuid=bundle["uuid"], version=bundle["version"], manifest=bundle["manifest"])
-    for f in bundle["files"]:
-        filename = f.pop("name")
-        if filename == "links.json":
-            links = f['body']['links']
-        file_row = File(**f)
-        bf_links.append(BundleFileLink(bundle=bundle_row, file=file_row, name=filename))
-    config.db_session.add_all(bf_links)
-    config.db_session.commit()
-    load_links(links)
+def combine_file_data(bundle):
+    combined_file_data = {}
+    for f in bundle["manifest"]['files']:
+        uuid = f['uuid']
+        combined_file_data[uuid] = f
+
+    for f in bundle['files']:
+        uuid = f['uuid']
+        combined_file_data[uuid] = {**f, **combined_file_data[uuid]}
+    for uuid in combined_file_data:
+        file = combined_file_data[uuid]
```

Comment:
It's generally not desirable to shadow built-in names. Usually people suffix an underscore to get around that: `file_`.

Reply:
👍